src/top_model/main.cpp

```diff
@@ -382,5 +384,11 @@ int main(int argc, char ** argv){
                    std::ratio<1>>>(hclock::now() - start).count();
                    ///< Run-time of simulation
     cout << "Simulation took:" << elapsed << "sec" << endl;
+
+    /**
+     * transform the output file generated by simulator to readable text
+     */
+    transform_output("data/top_model/abp_output.txt","data/top_model/abp_output_transform.txt");

```

Comment:
You can use softcoding for output and input file
using #define name "path to file"

You can do for every file

Reply:
I believe defining it using #define or supplying it directly as parameter is one and the same. Both ways can’t be called as soft coding.  

The only way to make code soft coding is to supply these paths through a configuration file wherein user can change the file paths easily by updating the settings file and without any need to update code.

Reply:
Even if developer uses #define to supply file paths, in future if someone wants to give a new file path, they have to update the code and do testing which is infeasible for users.

Reply:
so with #define user can easily change path at just one place in file instead of changing in output file and then in transform function call

Reply:
Yea. I agree but still it will be hard coding to me. As of now we can implement it using #define. But ill see if i can create a configuration file and make it more better

Reply:
Sorry i forgot to mention it was a suggestion no necessary change is needed.

Reply:
No problem. We will implement it anyways. Good catch btw 😊

Reply:
Changed to meet suggestion.